server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -194,14 +194,27 @@ public IndexNameExpressionResolver(Settings settings) {
             }
 
             Collection<IndexMetaData> resolvedIndices = aliasOrIndex.getIndices();
-            if (resolvedIndices.size() > 1 && !options.allowAliasesToMultipleIndices()) {
```

Comment:
given that this conditional is replaced, does that mean that allowAliasesToMultipleIndices is now always used together with the new requireAliasesToWriteIndex ? Or are there still cases where we don't allow aliases to multiple indices, yet we don't require aliases to write indices?

Reply:
if we do not allow multiple indices and do not require a write index... the normal resolution that occurs today where alias will resolve to the only index associated with it is still OK, even if it is not the write index.

Reply:
I am not following where that happens though, as the conditional for `allowAliasesToMultipleIndices` alone has become the conditional for `allowAliasesToMultipleIndices && requireAliasesToWriteIndex `. Do we have the old behaviour somewhere else which addresses the case where `allowAliasesToMultipleIndices && requireAliasesToWriteIndex ==false`?

Reply:
```allowAliasesToMultipleIndices && requireAliasesToWriteIndex ==false``` occurs during searching, no?

Reply:
argh, the exclamation mark bites again. I thought that the new if was the same as before plus a new condition. 
What still confuses me here though is that the old conditional `if (resolvedIndices.size() > 1 && !options.allowAliasesToMultipleIndices())` is gone as well as the `IllegalArgumentException` thrown in such case.  
What happens now when allowAliasesToMultipleIndices  is false? Does it never happen that allowAliasesToMultipleIndices is false? I kind of imply from this change that there are no API left with allowAliasesToMultipleIndices set to false? Do I read this correctly?

Reply:
get,mget do not allow aliases to multiple indices

Reply:
get,mget do not allow aliases to multiple indices

Reply:
ok I will dig and find out, too curious now :)